use CLI `output` value as path of `generate` directory (if an `output` value is provided)

keaton22 opened the pull request:
## Problem
- As far as I can tell, when opening a ZAP project from the CLI, if a value is provided for `--output`/`--out`/`--o`, that value does not get used as the default path for the `Generate` directory. I believe it should.



## Solution
- This PR makes the value of `--output`/`--out`/`--o` query-able from the front-end by passing it as a parameter in the `uiFunction()` in `startup.js`.

## Notes
- In Windows, the path value passed to `--output`/`--out`/`--o` must use escaped backslashes (e.g. `C:\\Users\\username\\zap-projects`)

keaton22 commented on src-electron/ui/window.js:
```diff
@@ -73,7 +73,8 @@ function createQueryString(
   restPort,
-  zapFileExtensions
+  zapFileExtensions,
+  output
```
I've updated the commit to fix this

keaton22 commented:
@paulr34 @tecimovic @brdandu I want to be respectful of your time and availability, but I would be very appreciative if I could get some feedback on this pull request. Thank you!

keaton22 commented:
@tecimovic @brdandu sorry to ping you both again but I wanted to follow up on this PR and get a second review / approval (and get the remaining GitHub actions kicked off as well). I rebased my branch against `master` earlier today so everything should be up to date.

Thank you again to @paulr34 for your earlier review & approval!

paulr34 commented:
> @tecimovic @brdandu sorry to ping you both again but I wanted to follow up on this PR and get a second review / approval (and get the remaining GitHub actions kicked off as well). I rebased my branch against `master` earlier today so everything should be up to date.
> 
> Thank you again to @paulr34 for your earlier review & approval!

catching up, @keaton22 the workflow still is not running, do you know why?

paulr34 commented:
> > @tecimovic @brdandu sorry to ping you both again but I wanted to follow up on this PR and get a second review / approval (and get the remaining GitHub actions kicked off as well). I rebased my branch against `master` earlier today so everything should be up to date.
> > Thank you again to @paulr34 for your earlier review & approval!
> 
> catching up, @keaton22 the workflow still is not running, do you know why?

I fixed it for you, @ethanzhouyc @dhchandw could you guys take a look at this PR and evaluate?

keaton22 commented:
> > > @tecimovic @brdandu sorry to ping you both again but I wanted to follow up on this PR and get a second review / approval (and get the remaining GitHub actions kicked off as well). I rebased my branch against `master` earlier today so everything should be up to date.
> > > Thank you again to @paulr34 for your earlier review & approval!
> > 
> > 
> > catching up, @keaton22 the workflow still is not running, do you know why?
> 
> I fixed it for you, @ethanzhouyc @dhchandw could you guys take a look at this PR and evaluate?

Thank you for getting the workflow running @paulr34! It looks like a "socket hang up" caused the Ubuntu build action to fail, hopefully it's a transient issue and can be fixed with a rerun.

paulr34 commented:
> > > > @tecimovic @brdandu sorry to ping you both again but I wanted to follow up on this PR and get a second review / approval (and get the remaining GitHub actions kicked off as well). I rebased my branch against `master` earlier today so everything should be up to date.
> > > > Thank you again to @paulr34 for your earlier review & approval!
> > > 
> > > 
> > > catching up, @keaton22 the workflow still is not running, do you know why?
> > 
> > 
> > I fixed it for you, @ethanzhouyc @dhchandw could you guys take a look at this PR and evaluate?
> 
> Thank you for getting the workflow running @paulr34! It looks like a "socket hang up" caused the Ubuntu build action to fail, hopefully it's a transient issue and can be fixed with a rerun.

strange, I can run the same job on a different PR and see
